### ISSUE TITLE
Compilation failure with ROCm 6

### ISSUE THREAD

**Issue by msimberg** [eth-cscs/DLA-Future#1239]
As reported by @mtaillefumier, DLA-Future 0.7.0 compilation fails with ROCm 6.0.3 (but not with 6.2.2) on LUMI:
```
 /tmp/mataille/spack-stage/spack-stage-dla-future-0.7.0-zjs6ptf4u22albgpidpcx2vqshkz6oix/spack-src/src/lapack/gpu/larft.cu:50:19: error: use of ove
            rloaded operator '*' is ambiguous (with operand types 'hipFloatComplex' (aka 'HIP_vector_type<float, 2>') and 'HIP_vector_type<float, 2>')
     124        t_ij = -tau_j * t_ij;
     125               ~~~~~~ ^ ~~~~
     126    /tmp/mataille/spack-stage/spack-stage-dla-future-0.7.0-zjs6ptf4u22albgpidpcx2vqshkz6oix/spack-src/src/lapack/gpu/larft.cu:121:12: note: in instant
            iation of function template specialization 'dlaf::gpulapack::kernels::fix_tau<32U, 32U, HIP_vector_type<float, 2>>' requested here
     127      kernels::fix_tau<kts_rows_t, kts_cols_t>
     128               ^
     129    /opt/rocm-6.0.3/lib/llvm/bin/../../../include/hip/amd_detail/amd_hip_vector_types.h:548:65: note: candidate function

     ...

     132    /opt/rocm-6.0.3/lib/llvm/bin/../../../include/hip/amd_detail/amd_hip_complex.h:239:1: note: candidate function
     133    COMPLEX_MUL_OP_OVERLOAD(hipFloatComplex)
     134    ^
     135    /opt/rocm-6.0.3/lib/llvm/bin/../../../include/hip/amd_detail/amd_hip_complex.h:80:40: note: expanded from macro 'COMPLEX_MUL_OP_OVERLOAD'
     136        __HOST_DEVICE__ static inline type operator*(const type& lhs, const type& rhs) {           \
     137                                           ^
  >> 138    /tmp/mataille/spack-stage/spack-stage-dla-future-0.7.0-zjs6ptf4u22albgpidpcx2vqshkz6oix/spack-src/src/lapack/gpu/larft.cu:50:19: error: use of ove
            rloaded operator '*' is ambiguous (with operand types 'hipDoubleComplex' (aka 'HIP_vector_type<double, 2>') and 'HIP_vector_type<double, 2>')
     139        t_ij = -tau_j * t_ij;
     140               ~~~~~~ ^ ~~~~
     141    /tmp/mataille/spack-stage/spack-stage-dla-future-0.7.0-zjs6ptf4u22albgpidpcx2vqshkz6oix/spack-src/src/lapack/gpu/larft.cu:121:12: note: in instant
            iation of function template specialization 'dlaf::gpulapack::kernels::fix_tau<32U, 32U, HIP_vector_type<double, 2>>' requested here
     142      kernels::fix_tau<kts_rows_t, kts_cols_t>
     143               ^
     144    /opt/rocm-6.0.3/lib/llvm/bin/../../../include/hip/amd_detail/amd_hip_vector_types.h:548:65: note: candidate function
```

This looks like the same issue we patched in https://github.com/eth-cscs/DLA-Future/pull/1195 for `add.cu`, but this one is in `larft.cu`, introduced in https://github.com/eth-cscs/DLA-Future/pull/1219/files. I would suggest we patch it up locally similarly to #1195 and release 0.7.1?

**Comment by msimberg** [comment]
https://github.com/eth-cscs/DLA-Future/pull/1240 with HIP 6.0.2 from dockerhub at least reproduces the issue: https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/4700071344751697/7514005670787789/-/jobs/8654956296#L231.

**Comment by msimberg** [comment]
Fixed by #1241.
